include/tvm/relay/op_attr_types.h

```diff
@@ -41,14 +41,17 @@ enum OpPatternKind {
   // for example :code:`out[i, ax1, j, ax2] = input[i, j]`.
   // Note that the axis need to be in order so transpose is not a bcast operator.
   kBroadcast = 1,
+  // The pattern for tuple nodes. Can fuse into subsequent injective ops.
+  kTuple = 2,
```

Comment:
I think we need to justify a bit why put kTuple as 2. Because tuple is special, I would rather put it say like 7, and use 
```
pattern <= kInjective || pattern == kTuple
```
to indicate the pattern is tuple aware

Reply:
I thought the pattern of tuple needs to be smaller than kInjective, because when we fuse the tuple into subsequent injective ops, we want the pattern of fused group to be kInjective ([CombinePattern](https://github.com/dmlc/tvm/blob/master/src/relay/pass/fuse_ops.cc#L578) returns the larger of the two patterns begin combined).  

But I realized that CombinePattern is only called when child group's [master ref is non null](https://github.com/dmlc/tvm/blob/master/src/relay/pass/fuse_ops.cc#L597), so kTuple doesn't have to be smaller than kInjective.

Reply:
It also means even if injective ops are fused into a broadcast op, the combined pattern is still kBroadcast. Is this intended?

Reply:
I see, never mind, we can keep this as it is then. Please add a comment on why we choose such order

Reply:
I've just changed the kTuple to 7, no change to fuse_ops.cc was needed.

Reply:
Let us update CombinePattern to specially handle tuple + injective and tuple + broadcast

Reply:
do you mean, both injective and broadcast win against tuple, even though their op pattern is smaller?

Reply:
@tqchen do you want to update both CombinePattern and its call site? (it seems CombinePattern is only called when one of its arg is kOutEWiseFusable.)